ocaml/idl/ocaml_backend/taskHelper.ml

```diff
@@ -176,7 +176,11 @@ let is_cancelling ~__context =
 	  List.exists (fun (_,x) -> x=`cancel) l
 
 let exn_if_cancelling ~__context =
-  if is_cancelling ~__context then raise (Api_errors.Server_error (Api_errors.task_cancelled,[]))
+	if is_cancelling ~__context
+	then begin
+		let task_id = Context.get_task_id __context in
+		raise Api_errors.(Server_error (task_cancelled, [Ref.string_of task_id]))
```

Comment:
exn_if_cancelling would be more reusable if it took a cleanup function - there are a few bits of code of the form

```
if is_cancelling(task):
  do_some_cleanup()
  raise task_cancelled
```

I'll see if I can tidy things up a bit.